server/integration/model.ts

```diff
@@ -0,0 +1,19 @@
+export default (sequelize, dataTypes) =>
+	sequelize.define(
+		'integration',
+		{
+			name: dataTypes.TEXT,
+			authSchemeName: dataTypes.ENUM('OAuth1'), // this frames polymorphism, add schema names here and in corresponding type
```

Comment:
What are your thoughts on simplifying this a bit? Maybe a user could have an association with a `ZoteroIntegration` that has an association to an `OAuth1Integration` table that stores Zotero's OAuth1-related state. This way we're still storing all OAuth1 integrations in the same table, but each specific integration get its own base table?

The reason I bring this up is because polymorphic relations like this in SQL seem kind of like an antipattern as we aren't able to enforce that an `Integration` can have at most a single `IntegrationData*` association.

Reply:
I'm tracking the thinking, but I believe we have the tooling to enforce the single `IntegrationData*` association.
the model for `Integration`:
```
...
authSchemeName: dataTypes.ENUM('OAuth1'),
...
integration.hasOne(integrationDataOAuth1);
...
```
the model for `IntegrationDataOAuth1`:
```
integrationDataOAuth1.belongsTo(integration, { scope: { authSchemeName: 'OAuth1'} });
```

If I'm getting this right, the Integration can have at most one IntegrationDataOAuth1, and the IntegrationDataOAuth1 can only be related to Integration models when they have the appropriate authSchemeName property. I read the intersection of those conditions as enforcement of 1:1 between `Integration` and `IntegrationData*`. Does that check out?

Reply:
quoting off-band communication with Eric to explain subsequent commits abandoning polymorphic approach:

>3mcd: AFAIK this doesn’t enforce data integrity at the database level but at the ORM level. I think the association scope `authSchemeName: ‘OAuth1’` boils down to a `WHERE` clause when Sequelize builds underlying queries to fetch associated models. It doesn’t enforce any kind of DB-level constraint (like a foreign key). I’m not sure if Sequelize would even consider the association scope if we accidentally added more than one `IntegrationData*` to an `Integration` instance/row. Your solution is definitely elegant and easy to understand, just thinking about the data integrity. If we’re cool with that, I’m okay with approving this as-is.